Samples/custom-transform-adapter/server.go

```diff
@@ -0,0 +1,284 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+
+package main
+
+import (
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"time"
+
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/date"
+	"github.com/google/uuid"
+	"github.com/gorilla/mux"
+	"github.com/iot-for-all/iotc-device-bridge/custom-transform-adapter/lib/bridge"
+	"github.com/mitchellh/mapstructure"
+	log "github.com/sirupsen/logrus"
+)
+
+const maxBodySize = 1024 * 1024 // 1 MiB
+
+type BridgeClient interface {
+	SetAuthorizer(autorest.Authorizer)
+	SetRetryAttempts(int)
+	SendMessage(context.Context, string, *bridge.MessageBody) (autorest.Response, error)
+	GetBaseURI() string
+}
+
+type BridgeClientAutorest struct {
+	bridge.BaseClient
+}
+
+func (client *BridgeClientAutorest) SetAuthorizer(authorizer autorest.Authorizer) {
+	client.Authorizer = authorizer
+}
+
+func (client *BridgeClientAutorest) SetRetryAttempts(retryAttempts int) {
+	client.RetryAttempts = retryAttempts
+}
+
+func (client *BridgeClientAutorest) SendMessage(ctx context.Context, deviceID string, body *bridge.MessageBody) (autorest.Response, error) {
+	return client.BaseClient.SendMessage(ctx, deviceID, body)
+}
+
+func (client *BridgeClientAutorest) GetBaseURI() string {
+	return client.BaseURI
+}
+
+type Adapter struct {
+	GetBridgeClient func() BridgeClient
+	Router          *mux.Router
+	Engine          *TransformEngine
+}
+
+// D2C message route definition augmented to include the Id of the cached transform queries.
+type AugmentedD2CMessage struct {
+	D2CMessage
+	TransformId         string
+	DeviceIdBodyQueryId string
+}
+
+// Builds a transform adapter for a given configuration.
+func NewAdapterFromConfig(config *Config, bridgeEndpoint string) *Adapter {
+	log.Info(fmt.Sprintf("Initializing adapter for Bridge %s", bridgeEndpoint))
+
+	adapter := Adapter{
+		Engine: NewTransformEngine(),
+		Router: mux.NewRouter(),
+		GetBridgeClient: func() BridgeClient {
+			return &BridgeClientAutorest{bridge.NewWithBaseURI(bridgeEndpoint)}
+		},
+	}
+
+	for _, message := range config.D2CMessages {
+		log.Info(fmt.Sprintf("Initializing route %s", message.Path))
+		augmentedMessage := AugmentedD2CMessage{D2CMessage: message}
+
+		// Initialize cache for request body transform.
+		if message.Transform != "" {
+			augmentedMessage.TransformId = uuid.New().String()
+			if err := adapter.Engine.AddTransform(augmentedMessage.TransformId, message.Transform); err != nil {
+				log.WithField("error", err).Panic(fmt.Sprintf("Failed to add request body transform for route %s: %s", message.Path, err))
```

Comment:
this is an antipattern. panics should be reserved for the very top level of the program. this should be returning an error which is handled at the higher level